Brokerages/Zerodha/ZerodhaSubscriptionManager.cs

```diff
@@ -0,0 +1,640 @@
+            tick.SellQuantity = ReadInt(b, ref offset);
+            tick.Open = ReadInt(b, ref offset) / divisor;
+            tick.High = ReadInt(b, ref offset) / divisor;
+            tick.Low = ReadInt(b, ref offset) / divisor;
+            tick.Close = ReadInt(b, ref offset) / divisor;
+
+            return tick;
+        }
+
+        /// <summary>
+        /// Reads a full mode tick from raw binary data
+        /// </summary>
+        private Tick ReadFull(byte[] b, ref int offset)
+        {
+            Tick tick = new Tick();
+            tick.Mode = Constants.MODE_FULL;
+            tick.InstrumentToken = ReadInt(b, ref offset);
+
+            decimal divisor = (tick.InstrumentToken & 0xff) == 3 ? 10000000.0m : 100.0m;
+
+            tick.Tradable = (tick.InstrumentToken & 0xff) != 9;
+            tick.LastPrice = ReadInt(b, ref offset) / divisor;
+            tick.LastQuantity = ReadInt(b, ref offset);
+            tick.AveragePrice = ReadInt(b, ref offset) / divisor;
+            tick.Volume = ReadInt(b, ref offset);
+            tick.BuyQuantity = ReadInt(b, ref offset);
+            tick.SellQuantity = ReadInt(b, ref offset);
+            tick.Open = ReadInt(b, ref offset) / divisor;
+            tick.High = ReadInt(b, ref offset) / divisor;
+            tick.Low = ReadInt(b, ref offset) / divisor;
+            tick.Close = ReadInt(b, ref offset) / divisor;
+
+            // KiteConnect 3 fields
+            tick.LastTradeTime = Utils.UnixToDateTime(ReadInt(b, ref offset));
+            tick.OI = ReadInt(b, ref offset);
+            tick.OIDayHigh = ReadInt(b, ref offset);
+            tick.OIDayLow = ReadInt(b, ref offset);
+            tick.Timestamp = Utils.UnixToDateTime(ReadInt(b, ref offset));
+
+
+            tick.Bids = new DepthItem[5];
+            for (int i = 0; i < 5; i++)
+            {
+                tick.Bids[i].Quantity = ReadInt(b, ref offset);
+                tick.Bids[i].Price = ReadInt(b, ref offset) / divisor;
+                tick.Bids[i].Orders = ReadShort(b, ref offset);
+                offset += 2;
+            }
+
+            tick.Offers = new DepthItem[5];
+            for (int i = 0; i < 5; i++)
+            {
+                tick.Offers[i].Quantity = ReadInt(b, ref offset);
+                tick.Offers[i].Price = ReadInt(b, ref offset) / divisor;
+                tick.Offers[i].Orders = ReadShort(b, ref offset);
+                offset += 2;
+            }
+            return tick;
+        }
+
+        protected override bool Subscribe(IEnumerable<Symbol> symbols, TickType tickType)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected override bool Unsubscribe(IEnumerable<Symbol> symbols, TickType tickType)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected override string ChannelNameFromTickType(TickType tickType)
+        {
+            throw new NotImplementedException();
+        }
```

Comment:
I think these need to be implemented?

The `ZerodhaSubscriptionManager` class seems based on our `BitfinexSubscriptionManager` implementation:
I would recommend checking it out since it was updated recently